### PR TITLE
Vlingo e2e sys airline ops UI work

### PR DESCRIPTION
**Pull request by abdullahcalisir12**

### CODE REVIEW

**Comment by VaughnVernon** [comment]
@Florian-Schoenherr and @danilo-ambrosio please review.

Current problems and incomplete implementations:

- monitoring and fleet crew endpoints have bugs
- fleet crew returns empty objects for get endpoint
- need an endpoint for monitoring
- maybe some clean up is needed (there was some copy and paste)

File: vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	<title>Fleet Crew</title>
+</svelte:head>
+
+<script>
+	import { Select, Button, Dialog, Row, Alert } from 'svelte-materialify/src';
+	import CardForm from '../components/CardForm.svelte';
+	import { Api } from "../api";
+	import { flights } from "../stores/flights.js";
+	import { inventories } from "../stores/inventory.js";
+	import { fleetcrews } from "../stores/fleetcrew.js";
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte:11]: Don't know why you use one file for each, just clutters imports in my opinion

File: vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	$: flightsForSelect = $flights.map((f => {
+		return {
+			name: f.id,
+			value: f.id,
+			...f
+		};
+	}));
+
+	$: {
+		if (selectedAircraft) {
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte:57]: I actually don't know, is there a difference or can you just do:
`$: if (selectedAircraft) { ... }` ?

File: vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	}
+
+	$: {
+		if (selectedFlight) {
+			const f = flightsForSelect.find(flight => flight.id === selectedFlight);
+			formData.flightNumber = f.id;
+		}
+	}
+
+	$: valid = !!selectedAircraft && !!selectedFlight;
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/routes/fleet-crew.svelte:72]: I like this :)

File: vlingo-e2e-sys-airline-ops/frontend/src/components/VlButton.svelte [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	export let text = "";
+	export let outlined = false;
+	export let icon = "";
+	export let href = "";
+	export let disabled = false;
+
+	function clicked() {
+		if (href) {
+			goto(href);
+			return;
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/components/VlButton.svelte:18]: return not needed

File: vlingo-e2e-sys-airline-ops/frontend/src/components/VlSelect.svelte [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<script>
+	import { Select } from 'svelte-materialify/src';
+	import { createEventDispatcher } from 'svelte';
+
+	const dispatch = createEventDispatcher();
+
+	export let value = "";
+
+  $: {
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/components/VlSelect.svelte:9]: like the other one

**Review by Florian-Schoenherr — CHANGES_REQUESTED** [review]
This is nice! The code is pretty svelte. Again, only minor suggestions.

File: vlingo-e2e-sys-airline-ops/frontend/src/components/CardForm.svelte [hunk 6]
```diff
@@ -16,6 +31,28 @@
...
+			<Alert class="error-color">
+				<div>
+					There is no {title}! Add one.
+				</div>
+			</Alert>
+		{/if}
+		{#if dialogFormPresent}
+			<Button on:click={toggleDialog}>{buttonText}</Button>
+			<Dialog persistent class="pa-8" bind:active={isDialogActive}>
+				<form on:submit|preventDefault={formSubmit}>
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/components/CardForm.svelte:44]: `<form on:submit|preventDefault>` works, so also delete formSubmit 👍

**Comment by abdullahcalisir12** [comment]
@Florian-Schoenherr PR is updated can you review it again?

File: vlingo-e2e-sys-airline-ops/frontend/src/routes/_layout.svelte [hunk 7]
```diff
@@ -22,7 +20,7 @@
 
 <svelte:window on:resize={isMobile.check} />
 
-<div style="height: 100vh; background-color: {bgTheme}">
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/routes/_layout.svelte]: Ah yes, I fixed that in materialify so on the other UI's you can delete this as well (or as we said, maybe we make a little component lib)

File: vlingo-e2e-sys-airline-ops/frontend/src/stores/index.js [hunk 8]
```diff
@@ -33,7 +33,7 @@ export function createLocalStore(key, initialValue) {
 	};
 }
 
-export const theme = createLocalStore('theme', 'light');
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/stores/index.js]: `createLocalStore` made sense, right? :)
probably the others will get some localstorage, too, so you can revert this one.

File: vlingo-e2e-sys-airline-ops/frontend/src/components/VlSelect.svelte [hunk 9]
```diff
@@ -6,11 +6,9 @@
 
 	export let value = "";
 
-  $: {
-    if (value) {
+  $: if (value) {
```
Comment (Florian-Schoenherr) [inline vlingo-e2e-sys-airline-ops/frontend/src/components/VlSelect.svelte:9]: just realized:
`$: dispatch('change', value);`
would even be enough! Should just put that into materialify itself.

**Review by Florian-Schoenherr — CHANGES_REQUESTED** [review]
Nice refactor.

File: vlingo-e2e-sys-airline-ops/frontend/src/stores/index.js (hunk 8 above)
Comment (abdullahcalisir12) [reply vlingo-e2e-sys-airline-ops/frontend/src/stores/index.js]: name is updated

**Review by Florian-Schoenherr — APPROVED** [review]

**Pull request merged**